src/main/java/ca/bc/hlth/mohorganizations/LoggingFilter.java

```diff
@@ -0,0 +1,42 @@
+package ca.bc.hlth.mohorganizations;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+@Component
+public class LoggingFilter extends OncePerRequestFilter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingFilter.class);
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+        int responseStatus = response.getStatus();
+        LOGGER.info("{} : {}", request.getMethod(), request.getRequestURI());
+        LOGGER.info("Completed : {}", responseStatus);
+
+        if(isSuccessfulRequest(responseStatus)){
```

Comment:
The LoggingFilter will be called before the controller is invoked so the response you're checking is just the initial response which will always have a status code of 200. You need to check the response after it's passed through the filters/controller. Something like this https://stackoverflow.com/questions/33744875/spring-boot-how-to-log-all-requests-and-responses-with-exceptions-in-single-pl.

Also, we need to add an ExceptionHandler otherwise unchecked exceptions will bypass the filter chain and the LoggingFilter will never be called.